src/init.cpp

```diff
@@ -1915,73 +1915,89 @@ bool AppInitMain(InitInterfaces& interfaces)
     if(gArgs.GetBoolArg("-gen", DEFAULT_GENERATE)) {
         LOCK(cs_main);
 
-        CTxDestination destination = DecodeDestination(gArgs.GetArg("-masternode_operator", ""));
-
-        CKeyID const operatorId = destination.which() == 1 ? CKeyID(*boost::get<PKHash>(&destination)) :
-                                  (destination.which() == 4 ? CKeyID(*boost::get<WitnessV0KeyHash>(&destination)) : CKeyID());
-        if (operatorId.IsNull()) {
-            LogPrintf("Error: wrong (or empty) masternode_operator address (%s)\n", gArgs.GetArg("-masternode_operator", "").c_str());
-            return false;
+        auto wallets = GetWallets();
+        if (wallets.size() == 0) {
+            LogPrintf("Warning! wallets not found\n");
+            return true;
         }
-        {
-            pos::ThreadStaker::Args stakerParams{};
-            {
-                std::vector<std::shared_ptr<CWallet>> wallets = GetWallets();
-                if (wallets.size() == 0) {
-                    LogPrintf("Warning! wallets not found\n");
-                    return true;
-                }
-                std::shared_ptr<CWallet> defaultWallet = wallets[0];
 
-                CKey minterKey;
-                bool found =false;
-                for (auto&& wallet : wallets) {
-                    if (wallet->GetKey(operatorId, minterKey)) {
-                        found = true;
-                        break;
-                    }
-                }
-                if (!found) {
-                    LogPrintf("Error: masternode operator private key not found\n");
-                    return false;
+        std::set<std::string> operatorsSet;
+        bool atLeastOneRunningOperator = false;
+        auto const operators = gArgs.GetArgs("-masternode_operator");
+
+        for (auto const & op : operators) {
```

Comment:
How does the setting of multiple operators look like in the conf now?

Can we have backwards compatibility with the current setting?

Reply:
Backward compatibility is shown in functional test. The settings is same as cli settings you just specify multiple options
`defid -masternode_operator=<fist-one> -masternode_operator=<second-one>`